docs/content/sips/000-spin-test.md

```diff
@@ -0,0 +1,189 @@
+    /// A handle into an interface that tracks calls to the key-value store
+    import fermyon:spin-test-virt/key-value-calls;
+    /// TODO: more handles to configure and observe the other Spin
+    /// and WASI interfaces.
+    
+    /// The following allow the test to make requests against the Spin app
+    /// and view its response.
+    import http-helper;
+    /// The ability to call the Spin app
+    import wasi:http/incoming-handler@0.2.0;
+
+	/// Actually call the test
+    export run: func();
+}
+
+interface http-helper {
+    use wasi:http/types@0.2.0.{incoming-request, response-outparam, incoming-response, outgoing-request};
+    resource response-receiver {
+        get: func() -> option<incoming-response>;
+    }
+    new-request: func(request: outgoing-request) -> incoming-request;
+    new-response: func() -> tuple<response-outparam, response-receiver>;
+}
+```
+
+### The `fermyoin:spin-test-virt/plug` world
+
+This is the world of the component that virtualizes a Spin runtime. 
+
+```
+package fermyon:spin-test-virt;
+
+/// The exports that can be composed with a Spin app creating
+/// a virtualized component.
+world plug {
+	/// All of the `fermyon:spin/platform` world's interfaces are
+	/// exported here.
+    export fermyon:spin/key-value@2.0.0;
+    export fermyon:spin/llm@2.0.0;
+    export fermyon:spin/redis@2.0.0;
+    export fermyon:spin/postgres@2.0.0;
+    export fermyon:spin/mqtt@2.0.0;
+    export fermyon:spin/mysql@2.0.0;
+    export fermyon:spin/sqlite@2.0.0;
+    export fermyon:spin/variables@2.0.0;
+    export wasi:http/outgoing-handler@0.2.0;
+    
+    /// The virtualization needs the component's id and
+    /// the `spin.toml` manifest because many of the spin
+    /// interfaces are configured through that combination.
+    export set-component-id: func(component-id: string);
+    import get-manifest: func() -> string;
+
+	/// A way to say that a certain URL is associated with a response
+    export http-handler;
+    /// How the calls to the kv store are tracked
+    export key-value-calls;
+    
+}
+
+interface http-handler {
+    use wasi:http/types@0.2.0.{future-incoming-response};
+    set-response: func(url: string, response: future-incoming-response);
+}
+
+interface key-value-calls {
+    reset: func();
+    get: func() -> list<tuple<string, list<get-call>>>;
+    set: func() -> list<tuple<string, list<set-call>>>;
+
+    record get-call {
+        key: string
+    }
+
+    record set-call {
+        key: string,
+        value: list<u8>
+    }
+}
+```
+
+### The `fermyoin:spin-test/runner` world
+
+```
+world runner {
+	/// The host must supply a manifest
+    import get-manifest: func() -> string;
+    /// The host must give the runner the ability to create HTTP resources
+    import http-helper;
+    /// The host can then run the test
+    export run: func();
+}
+```
+
+## Open Questions
+
+- Should `spin-test` be a stand alone CLI experience, a plugin for `spin` CLI, or both?
```

Comment:
I would say plugin is probably best - at least as a starting point - as it would benefit from the plugin update framework.

Reply:
+1 on the plugin route for delivering the functionality to users.